Framework/Algorithms/test/SumOverlappingTubesTest.h

```diff
@@ -25,6 +26,41 @@ using namespace Mantid::Indexing;
 using namespace Mantid::Kernel;
 using namespace Mantid::Types::Core;
 
+namespace {
+MatrixWorkspace_sptr createTestScanningWS(size_t nTubes, size_t nPixelsPerTube,
+                                          std::vector<double> rotations,
+                                          std::string name = "testWS") {
+  const auto instrument = ComponentCreationHelper::createInstrumentWithPSDTubes(
+      nTubes, nPixelsPerTube, true);
+  size_t nTimeIndexes = rotations.size();
+  size_t nBins = 1;
+
+  std::vector<std::pair<DateAndTime, DateAndTime>> timeRanges;
+  for (size_t i = 0; i < nTimeIndexes; ++i)
+    timeRanges.push_back(std::make_pair(DateAndTime(i), DateAndTime(i + 1)));
+
+  ScanningWorkspaceBuilder builder(instrument, nTimeIndexes, nBins);
+  builder.setTimeRanges(timeRanges);
+  builder.setRelativeRotationsForScans(rotations, V3D(0, 0, 0), V3D(0, 1, 0));
+
+  Points x(nBins, LinearGenerator(0.0, 1.0));
+  Counts y(std::vector<double>(nBins, 2.0));
+  builder.setHistogram(Histogram(x, y));
+
+  auto testWS = builder.buildWorkspace();
+
+  // This has to be added to the ADS so that it can be used with the string
+  // validator used in the algorithm.
+  AnalysisDataService::Instance().add(name, testWS);
```

Comment:
It's a real shame you have to do this, and I can see the `ADSValidator` on the input workspaces. @LamarMoore was working on a `WorkspaceListProperty` to try to rectify this exact situation. I don't recall what the status of that is now.

Reply:
I just checked with him, unfortunately it is not ready yet, but something to bear in mind for future tidy-up.

Reply:
Of course, this would have to be done carefully. Once it makes a release, we have to assume that users have scripts that use the current property mechanism.

Reply:
I was expecting a `WorkspaceListProperty` would look the same to the user as the current mechanism in Python, perhaps this would not be the case though?